requirements.txt

```diff
@@ -1,23 +1,40 @@
-Pyomo>=6.7.1
-numpy<2.0.0
-dill>=0.3.8
-pandas>=2.2.2
-requests>=2.31.0
-timezonefinder>=6.5.0
-statsmodels>=0.14.1
-pvlib>=0.10.4
-scipy>=1.13.0
-setuptools>=69.2.0
-matplotlib>=3.8.4
-openpyxl>=3.1.2
-pytest>=8.1.1
-scikit-learn>=1.4.2
-pwlf>=2.2.1
-gurobipy>=11.0.1
-scandir>=1.10.0
-tables>=3.9.2
-pre-commit>=3.7.0
-black>=24.4.0
-tsam>=2.3.1
-sphinx>=7.3.7
-sphinx_rtd_theme>=2.0.0
+certifi==2024.8.30 ; python_version >= "3.12" and python_version < "4.0"
+cffi==1.17.1 ; python_version >= "3.12" and python_version < "4"
+charset-normalizer==3.4.0 ; python_version >= "3.12" and python_version < "4.0"
+colorama==0.4.6 ; python_version >= "3.12" and python_version < "4.0" and (sys_platform == "win32" or platform_system == "Windows")
+et-xmlfile==2.0.0 ; python_version >= "3.12" and python_version < "4.0"
+h3==4.1.2 ; python_version >= "3.12" and python_version < "4"
+h5py==3.12.1 ; python_version >= "3.12" and python_version < "4.0"
+highspy==1.8.1 ; python_version >= "3.12" and python_version < "4.0"
+idna==3.10 ; python_version >= "3.12" and python_version < "4.0"
+iniconfig==2.0.0 ; python_version >= "3.12" and python_version < "4.0"
+joblib==1.4.2 ; python_version >= "3.12" and python_version < "4.0"
+networkx==3.4.2 ; python_version >= "3.12" and python_version < "4.0"
+numpy==2.1.3 ; python_version >= "3.12" and python_version < "4.0"
+openpyxl==3.1.5 ; python_version >= "3.12" and python_version < "4.0"
+packaging==24.2 ; python_version >= "3.12" and python_version < "4.0"
+pandas==2.2.3 ; python_version >= "3.12" and python_version < "4.0"
+patsy==1.0.1 ; python_version >= "3.12" and python_version < "4.0"
+pluggy==1.5.0 ; python_version >= "3.12" and python_version < "4.0"
+ply==3.11 ; python_version >= "3.12" and python_version < "4.0"
+pvlib==0.11.1 ; python_version >= "3.12" and python_version < "4.0"
+pwlf==2.3.0 ; python_version >= "3.12" and python_version < "4.0"
+pycparser==2.22 ; python_version >= "3.12" and python_version < "4"
+pydoe==0.3.8 ; python_version >= "3.12" and python_version < "4.0"
+pyomo==6.8.2 ; python_version >= "3.12" and python_version < "4.0"
+pytest==8.3.3 ; python_version >= "3.12" and python_version < "4.0"
+python-dateutil==2.9.0.post0 ; python_version >= "3.12" and python_version < "4.0"
+pytz==2024.2 ; python_version >= "3.12" and python_version < "4.0"
+requests==2.32.3 ; python_version >= "3.12" and python_version < "4.0"
+scandir==1.10.0 ; python_version >= "3.12" and python_version < "4.0"
+scikit-learn==1.5.2 ; python_version >= "3.12" and python_version < "4.0"
+scipy==1.14.1 ; python_version >= "3.12" and python_version < "4.0"
+setuptools==75.5.0 ; python_version >= "3.12" and python_version < "4"
```

Comment:
does it matter that it is 4 or 4.0?

Reply:
Nope, python 4 doesnt even exist :D, I guess its just added in case does exist at some point.